### PR TITLE
add kustomize overlay for running on mac1.metal

### PR DESCRIPTION
**Pull request by hidalgopl**

### CODE REVIEW

File: deploy/terraform-aws-mac/cleanup-vpc.sh [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        "Name=vpc-id,Values=${1}" |
+    jq -r '.Reservations | .[] | .Instances | .[] | .InstanceId'
+}
+
+instance_ids=$(get_pending_and_running_instance_ids_by_vpc_id "$VPC_ID")
+while [[ -n "$instance_ids" ]]
+do
+    echo "Terminating instances: $instance_ids"
+    aws ec2 terminate-instances --instance-ids "$instance_ids"
+    aws ec2 wait instance-terminated --instance-ids "$instance_ids"
```
Comment (henryprecheur) [inline deploy/terraform-aws-mac/cleanup-vpc.sh:60]: I believe this needs to be updated from master. This is my old broken code.

File: deploy/manifests/kip/overlays/mac/cloudinit.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+sudo /usr/local/bin/anka license activate -f $ANKA_LICENSE
+sudo /usr/local/bin/anka license accept-eula || true
+sudo /usr/local/bin/anka license validate
+sudo curl -o /usr/local/bin/itzo https://itzo-kip-dev-download.s3.amazonaws.com/itzo-darwin-820
+sudo chmod +x /usr/local/bin/itzo
+sudo /usr/local/bin/itzo -v=2 > /var/log/itzo.log 2>&1 &
+
+
```
Comment (ldx) [inline deploy/manifests/kip/overlays/mac/cloudinit.sh:9]: We should be able to use a regular YAML cloud-init file with the latest AMI, have you tried that?

File: deploy/manifests/kip/overlays/mac/anka_svc.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    kompose.cmd: kompose convert -o registry-manifests
+    kompose.version: 1.22.0 (955b78124)
+  labels:
+    io.kompose.service: anka-registry
```
Comment (ldx) [inline deploy/manifests/kip/overlays/mac/anka_svc.yaml:8]: What are all the "kompose" labels/annotations?

File: deploy/manifests/kip/overlays/mac/provider.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    accessKeyID: ""
+    secretAccessKey: ""
+etcd:
+  internal:
+    dataDir: /opt/kip/data
+cells:
+  bootImageSpec:
+    owners: 689494258501
+    filters: name=elotl-kip-*
+    macfilters: name=elotl-kipmac-*
```
Comment (ldx) [inline deploy/manifests/kip/overlays/mac/provider.yaml:13]: Did the multi-image PR end up using `macfilters`? That will be a problem once ARM Mac instances become available.

File: deploy/manifests/kip/overlays/mac/provider.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  defaultInstanceType: "t3.nano"
+  defaultVolumeSize: "20G"
+  nametag: kip
+  standbyCells:
+    - instanceType: "mac1.metal"
+      count: 1
+      spot: false
```
Comment (ldx) [inline deploy/manifests/kip/overlays/mac/provider.yaml:20]: We should rethink if we want to use standby instances at this point. The way they work, kip will allocate and start a new one if a pod gets bound to an existing one (basically headroom). This is good for reducing wait times, but while testing, it'll create wastage.

File: deploy/manifests/kip/overlays/mac/cloudinit.sh (hunk 2 above)
Comment (hidalgopl) [reply deploy/manifests/kip/overlays/mac/cloudinit.sh]: You're right, synced `cloudinit.yaml` & `provider.yaml` with latest changes :+1:

File: deploy/manifests/kip/overlays/mac/anka_svc.yaml (hunk 3 above)
Comment (hidalgopl) [reply deploy/manifests/kip/overlays/mac/anka_svc.yaml]: I generated those manifests from docker-compose using [kompose](https://kompose.io/). Changed to recommended k8s labels

File: deploy/manifests/kip/overlays/mac/provider.yaml (hunk 5 above)
Comment (hidalgopl) [reply deploy/manifests/kip/overlays/mac/provider.yaml]: :+1:  removing for now

File: deploy/manifests/kip/overlays/mac/provider.yaml (hunk 4 above)
Comment (henryprecheur) [reply deploy/manifests/kip/overlays/mac/provider.yaml]: This will have to be changed to x86_64_mac_filters. See #203

**Comment by hidalgopl** [comment]
closing this for now. For at least a while we gonna use anka registry running directly on mac host, so we don't need the manifests to run it as a part of the cluster. We can reopen it if needed

**Pull request closed without merging**